### ISSUE TITLE
Vyper version 0.2.10 not installed via brownie

### ISSUE THREAD

**Issue by skellet0r** [eth-brownie/brownie#964]
### Environment information

* `brownie` Version: 1.13.1
* `ganache-cli` Version: 6.12.2
* Python Version: 3.9.1
* OS: linux

### What was wrong?
Running `brownie compile` or `brownie test` errors out with `IncompatibleVyperVersion` exception.

> tl;dr: quick fix for users, just reinstall your venv, or upgrade vyper to the latest version using pip.

*contracts/Foo.vy*
```python
# @version 0.2.10

@external
def main() -> uint256:
    return 0
```

*Error Output*
```bash
$ brownie compile
Brownie v1.13.1 - Python development framework for Ethereum

  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/compile.py", line 50, in main
    proj = project.load()
  File "brownie/project/main.py", line 727, in load
    return Project(name, project_path)
  File "brownie/project/main.py", line 175, in __init__
    self.load()
  File "brownie/project/main.py", line 228, in load
    self._compile(changed, self._compiler_config, False)
  File "brownie/project/main.py", line 90, in _compile
    build_json = compiler.compile_and_format(
  File "brownie/project/compiler/__init__.py", line 97, in compile_and_format
    find_vyper_versions(vyper_sources, install_needed=True, silent=silent)
  File "brownie/project/compiler/vyper.py", line 137, in find_vyper_versions
    raise IncompatibleVyperVersion(
IncompatibleVyperVersion: No installable vyper version matching '0.2.10' in 'contracts/Test.vy'
```
This error appears for version `0.2.10` and for version `0.2.9`.

### How can it be fixed?

~Currently looking into this bug, but it looks like after the version `0.2.10` release of vyper today, the bug appeared. The bug smells like it's coming from the upstream vyperlang/vvm package.~

Looks like just upgrading the vyper version installed when installing eth-brownie fixes it. This goes double for if you have brownie installed in a virtualenv or globally via pipx.

## Update: Why does this happen

*vvm/install.py*
```python
def get_installable_vyper_versions(headers: Dict = None) -> List[Version]:
    """
    Return a list of all `vyper` versions that can be installed by vvm.

    Returns
    -------
    List
        List of Versions objects of installable `vyper` versions.
    """
    version_list = []

    headers = _get_headers(headers)

    for release in _get_releases(headers):
        version = Version.coerce(release["tag_name"].lstrip("v"))
        asset = next((i for i in release["assets"] if _get_os_name() in i["name"]), False)
        if asset:
            version_list.append(version)
    return sorted(version_list, reverse=True)
```

This piece of code queries the following endpoint `https://api.github.com/repos/vyperlang/vyper/releases?per_page=100` via the `_get_releases(...)` call. In return it gets a response with all of the vyper releases over at vyperlang/vyper. However for each release, if the release doesn't contain a precompiled vyper binary for the users os, it will correctly skip over that release (because there isn't a binary to download and use). Thus reurning a list of only versions with an asset like such: `vyper.0.2.8+commit.069936f.darwin`, the suffix being either linux, darwin, or windows.exe.

This is important because later on in the below snippet of code, that list is used as the `available_versions` variable.

*brownie/compiler/vyper.py::find_vyper_versions*
```python
def find_vyper_versions(
    contract_sources: Dict[str, str],
    install_needed: bool = False,
    install_latest: bool = False,
    silent: bool = True,
) -> Dict:

    """
    Analyzes contract pragmas and determines which vyper version(s) to use.

    Args:
        contract_sources: a dictionary in the form of {'path': "source code"}
        install_needed: if True, will install when no installed version matches
                        the contract pragma
        install_latest: if True, will install when a newer version is available
                        than the installed one
        silent: set to False to enable verbose reporting

    Returns: dictionary of {'version': ['path', 'path', ..]}
    """

    available_versions, installed_versions = _get_vyper_version_list()

    pragma_specs: Dict = {}
    to_install = set()
    new_versions = set()

    for path, source in contract_sources.items():
        pragma_specs[path] = sources.get_vyper_pragma_spec(source, path)  # This is an semver spec aware object, see semantic_version/base.py::BaseSpec
        version = pragma_specs[path].select(installed_versions)  # Select the best compatible version from installed versions else return None. This will equal None

        if not version and not (install_needed or install_latest):  # we called this function with install_needed=True so we can skip
            raise IncompatibleVyperVersion(
                f"No installed vyper version matching '{pragma_specs[path]}' in '{path}'"
            )

        # if no installed version of vyper matches the pragma, find the latest available version
        latest = pragma_specs[path].select(available_versions) # Select the best compatible version from available versions

        if not version and not latest:  # since 0.2.10 is not in the available versions list, latest = None
            raise IncompatibleVyperVersion(
                f"No installable vyper version matching '{pragma_specs[path]}' in '{path}'"
            )
...
```  

So basically **tl;dr**, vyperlang/vyper just needs to have the precompiled vyper assets in each release, otherwise users with a pre-existing environment won't be able to acquire newer versions of vyper via brownie and vvm.
